### DISCUSSION TITLE
I have read-only access, I can't send a request..

### DISCUSSION THREAD

**Discussion by sotoniak** [alandtse/tesla#797] [Q&A] [answered]
Hi. Does anyone have a hint as to why read only operations work for reading data from the car, but when I try to set any value it ends up with an error?

**Comment by IanK6449** [comment]
I have the same problem. I can only view whether the charger entity is on or off but I can't control it. I would like to be able to do this so I can switch charging on or off depending on how much solar power we're generating. I am not sure though whether the integration will allow this as the switches look to be listed as 'diagnostic'?

**Comment by JoseMadbird** [comment]
Hello, same here, i have a new M3 Highland, and not possible to write, just read......

**Comment by skipishere** [comment, marked as answer]
Hi all,

If you have an issue it would help us if you could specify some basic information such as the version of the extension, what car model you have and the firmware version your car is running. You might all be being affected by the change that Tesla has been working on that requires the use of the new API and newer cars can't use the old one (and older cars from next year) - https://github.com/alandtse/tesla/issues/743

![image](https://github.com/alandtse/tesla/assets/26724777/723426c0-4160-4150-86b5-13f9a8064fbc)

**Comment by IanK6449** [comment]
Hi Stephen,

Yeah that’s all good advice.
I will provide that but at this point I was more just curious as to whether that ‘switch’ should indeed function as a switch and be able to control the charger on the car?
From the Github description it sounded like it may be more just a monitor than something that enables actual control?

Regarding your questions:
Tesla:                                  Model Y 2023 model (built around August I think)
Firmware:                          2023.44.1 d0b7099fb39d
Integration version:         (HACS)  v3.19.3

Hope this helps and thank you to everybody who contributes to this integration… it must be a complex API given how many functions and sensors are available.
Cheers,

IK
